libs/langchain/langchain/chains/combine_documents/map_reduce.py

```diff
@@ -1,18 +1,197 @@
+    """Create a chain that updates the contents of a list of documents by passing them to a model.
+
+    Args:
+        llm: Language model to use for mapping document contents.
+        prompt: Prompt to use for mapping document contents. Must accept "context" as
+            one of the input variables. Each document will be passed in as "context".
+        document_prompt: Prompt used for formatting each document into a string. Input
+            variables can be "page_content" or any metadata keys that are in all
+            documents. "page_content" will automatically retrieve the
+            `Document.page_content`, and all other inputs variables will be
+            automatically retrieved from the `Document.metadata` dictionary. Default to
+            a prompt that only contains `Document.page_content`.
+
+    Returns:
+        An LCEL `Runnable` chain.
+
+        Expects a dictionary as input. Input must contain "context" key with a list of
+        Documents.
+
+        Returns a list of Documents, with the contents of each document being the output
+        of passing the corresponding input document to the model. Document order is
+        preserved.
+
+    Example:
+        .. code-block:: python
+
+            from langchain_community.chat_models import ChatOpenAI
+            from langchain_core.documents import Document
+            from langchain_core.prompts import ChatPromptTemplate
+            from langchain.chains.combine_documents import create_map_documents_chain
+
+            llm = ChatOpenAI(model="gpt-3.5-turbo")
+            extract_prompt = ChatPromptTemplate.from_template(
+                [
+                    ("system", "Given a user question, extract the most relevant parts of the following context:\n\n{context}"),
+                    ("human", "{question}"),
+                ]
+            )
+            map_documents_chain = create_map_documents_chain(llm, extract_prompt)
+
+            docs = [
+                Document(page_content="Jesse loves red but not yellow"),
+                Document(page_content = "Jamal loves green but not as much as he loves orange")
+            ]
+
+            map_documents_chain.invoke({"context": docs, "question": "Who loves green?"})
+    """  # noqa: E501
+    validate_prompt(prompt, (DOCUMENTS_KEY,))
+    _document_prompt = document_prompt or PromptTemplate.from_template("{page_content}")
+
+    _format = partial(format_document_inputs, document_prompt=_document_prompt)
+    map_content_chain = _format | prompt | llm | StrOutputParser()
+    map_content_chain = map_content_chain.with_config(run_name="map_content")
+
+    assign_page_content: Runnable = RunnablePassthrough.assign(
+        page_content=map_content_chain
+    )
+    assign_page_content = assign_page_content.with_config(
+        run_name="assign_page_content"
+    )
+    map_doc_chain = assign_page_content | _compile_document
+    map_docs_chain = map_doc_chain.with_config(run_name="map_document").map()
```

Comment:
i'd do the map when you use it in the expression, it doesnt really make the chain